### PR TITLE
platform(general): Add lines to SBOM 

### PR DESCRIPTION
**Pull request by NoaAzoulay**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/output/cyclonedx.py [hunk 1]
```diff
@@ -188,6 +188,10 @@ def create_library_component(self, resource: Record | ExtraResource, check_type:
           <name>flask</name>
           <version>0.6</version>
           <purl>pkg:pypi/cli_repo/pd/requirements.txt/flask@0.6</purl>
+          <properties>
+            <property name="startLine">5</property>
```
Comment (AdamDev) [inline checkov/common/output/cyclonedx.py:192]: What are these numbers? 5 and 6?

Comment (NoaAzoulay) [reply checkov/common/output/cyclonedx.py:192]: 5 is the start line number, 6 is the end line number (this is just an example)

**Review by AdamDev — APPROVED** [review]

**Review by itai1357 — APPROVED** [review]
Great job!

just pay attention to add test for the case of cvs report

File: checkov/common/output/common.py [hunk 2]
```diff
@@ -52,3 +53,9 @@ def format_string_to_licenses(licenses_str: str) -> list[str]:
 def compare_table_items_severity(table_item: dict[str, str]) -> int:
     severity = (table_item.get("severity") or DEFAULT_SEVERITY).upper()
     return Severities[severity].level
+
+
+def validate_lines(lines: list[int] | None) -> list[int] | None:
+    if lines and len(lines) > 0 and lines[0] > 0 and lines[1] > 0:
```
Comment (ayajbara) [inline checkov/common/output/common.py:59]:
```suggestion
    if lines and lines[0] > 0 and lines[1] > 0:
```
if the length is 0 the first condition will be false

**Review by gruebel — APPROVED** [review]
🍰

**Pull request merged**